### PR TITLE
feat: deploy entrypoint

### PR DESCRIPTION
**Pull request by souradeep-das**
:clipboard: associated with #504 

## Overview

Deploys AA contracts through a separate aa_deployer container - so that - AA features can start up with the stack when required

## Changes

- moves BundlerHelper (all AA) contracts under /account-abstraction
- adds aa_deployer to the stack and deploys contracts
- hosts addresses on the dtl

## Testing
AA unit-tests run on ci
check ci logger for contract deployment through aa_deployer
integration tests with the deployment to follow

### CODE REVIEW

File: ops/docker-compose.yml [hunk 1]
```diff
@@ -441,6 +441,31 @@ services:
     ports:
       - ${FRAUD_CHECKER_HTTP_PORT:-8555}:8555
 
+  # deploys account abstraction contracts and serves contract addresses
```
Comment (souradeep-das) [inline ops/docker-compose.yml]: do we want to move this to side? or spin with the main stack

Comment (InoMurko) [reply ops/docker-compose.yml]: side!

File: ops/docker/Dockerfile.aa_deployer [hunk 2]
```diff
@@ -0,0 +1,38 @@
+FROM bobanetwork/builder AS builder
```
Comment (InoMurko) [inline ops/docker/Dockerfile.aa_deployer:1]: just in about flux is the removal of the builder image, might wait a little bit to get that in?

File: packages/boba/account-abstraction/bin/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { Wallet, providers } from 'ethers'
+import { getContractFactory } from '@eth-optimism/contracts'
+
+/* eslint-disable */
+require('dotenv').config()
+
+import hre from 'hardhat'
+
+const main = async () => {
+  console.log('Starting BOBA core contracts deployment...')
```
Comment (InoMurko) [inline packages/boba/account-abstraction/bin/deploy.ts]:
```suggestion
  console.log('Starting BOBA AA contracts deployment...')
```

File: ops/docker-compose.yml (hunk 1 above)
Comment (souradeep-das) [reply ops/docker-compose.yml]: okey dokey

File: ops/docker/Dockerfile.aa_deployer (hunk 2 above)
Comment (souradeep-das) [reply ops/docker/Dockerfile.aa_deployer:1]: totally!

**Review by InoMurko — APPROVED** [review]
LGTM!

File: ops/docker-compose-side.yml [hunk 4]
```diff
@@ -85,6 +89,32 @@ services:
...
+      context: ..
+      dockerfile: ./ops/docker/Dockerfile.packages
+      target: aa_deployer
+    environment:
+      L1_NODE_WEB3_URL: http://l1_chain:8545
+      L2_NODE_WEB3_URL: http://l2geth:8545
+      ADDRESS_MANAGER_ADDRESS: "0x5FbDB2315678afecb367f032d93F642f64180aa3"
+      URL: http://dtl:8081/addresses.json
+      BOBA_URL: http://dtl:8081/boba-addr.json
+      # DO NOT use in production
```
Comment (wsdt) [inline ops/docker-compose-side.yml:110]: We could work with Env Interpolation here to distringuish between prod/dev variables.

**Review by wsdt — APPROVED** [review]
looks good to me, left a comment.

File: ops/docker-compose-side.yml (hunk 4 above)
Comment (souradeep-das) [reply ops/docker-compose-side.yml:110]: true, maybe we do that for all containers together? sticking to the current setup for now?

**Pull request merged**